src/main/scala/com/kinja/amqp/AmqpClientFactory.scala

```diff
@@ -127,4 +131,22 @@ class AmqpClientFactory {
 			config.askTimeOut
 		)(ec)
 	}
+
+	/**
+	 * Call disconnect on all created and tracked client.
+	 */
+	def disconnectAllClient(): Unit = clients.foreach(_.disconnect())
+
+	/**
+	 * Call reconnect on all created and tracked client.
+	 */
+	def reconnectAllClient(): Unit = clients.foreach(_.reconnect())
+
+	/**
+	 * Remove client from tracking.
+	 * @param client client to remove from tracking
+	 */
+	def removeClient(client: AmqpConsumerClientInterface): Unit = this.synchronized {
```

Comment:
Not that there is something wrong with it, but why do we need it?

Reply:
Currently we don't need it. But I can imagine situation when we want to have just a short living client which do it job and after that it stop itself.